zproject/backends.py

```diff
@@ -352,6 +352,7 @@ def authenticate(self, request: HttpRequest, *,
         if user_profile is None:
             return None
         if user_profile.check_password(password):
+            request.session['needs_to_change_password'] = not check_password_strength(password)
```

Comment:
Hmm, let's not set a session variable unless necessary - so only if the password is too weak, because it will be very rare - and doing this might generate an additional database, or at least cache, write.

Reply:
makes sense to me. Changed.

Reply:
We need to clear the session variable if it's set, though, right?